sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_models.py

```diff
@@ -513,104 +513,104 @@ def _from_certificate_operation_bundle(cls, certificate_operation_bundle):
 
     @property
     def id(self):
-        # type: () -> str
-        """:rtype: str"""
+        # type: () -> Optional[str]
+        """:rtype: str or None"""
         return self._id
 
     @property
     def name(self):
-        # type: () -> str
-        """:rtype: str"""
-        return self._vault_id.name
+        # type: () -> Optional[str]
+        """:rtype: str or None"""
+        return self._vault_id.name if self._vault_id else None
 
     @property
     def vault_url(self):
-        # type: () -> str
+        # type: () -> Optional[str]
         """URL of the vault containing the CertificateOperation
 
-        :rtype: str
+        :rtype: str or None
         """
-        return self._vault_id.vault_url
+        return self._vault_id.vault_url if self._vault_id else None
 
     @property
     def issuer_name(self):
-        # type: () -> str
+        # type: () -> Union[str, WellKnownIssuerNames, None]
         """The name of the issuer of the certificate.
 
-        :rtype: str
+        :rtype: str or WellKnownIssuerNames or None
```

Comment:
I would use the fully qualified name of WellKnownIssuerNames here: `~azure.keyvault.certificates.WellKnownIssuerNames` ([example](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py#L384) from `azure-keyvault-keys`). For the type hint on line 537, we can probably also do `Union[WellKnownIssuerNames, None]` because WellKnownIssuerNames is a string enum, and that lets the user know that the value will only be from that enum's set of values.

Reply:
I think we should keep `str` in Union, because on line496 the right side of "=" is type of Optional[str].